user-operator/src/main/java/io/strimzi/operator/user/operator/KafkaUserOperator.java

```diff
@@ -205,25 +207,34 @@ public final CountDownLatch reconcileAll(String trigger, String namespace, Label
                 .collect(Collectors.toSet());
         log.debug("reconcileAll({}, {}): Other resources with labels {}: {}", RESOURCE_KIND, trigger, resourceSelector, resourceNames);
 
-        Set<String> usersWithAcls = aclOperations.getUsersWithAcls();
-        log.debug("reconcileAll({}, {}): User with ACLs: {}", RESOURCE_KIND, trigger, usersWithAcls);
+        CountDownLatch outerLatch = new CountDownLatch(1);
 
-        desiredNames.addAll(resourceNames);
-        desiredNames.addAll(usersWithAcls);
+        vertx.createSharedWorkerExecutor("kubernetes-ops-pool").executeBlocking(
+            future -> {
+                Set<String> usersWithAcls = aclOperations.getUsersWithAcls();
+                future.complete(usersWithAcls);
+            }, res -> {
+                log.debug("reconcileAll({}, {}): User with ACLs: {}", RESOURCE_KIND, trigger, res.result());
```

Comment:
All this logic should be in a `if (res.succeeded())` and in the `else` you should log an error, I suppose.